### PR TITLE
Improve error handling

### LINKED ISSUE #4150: When you don't have enableSemantics() an ugly exception appears
### Setup and configuration

- SMW version: master
- MW version: 1.27
- PHP version: 7.3
- DB system (MySQL, Blazegraph, etc.) and version: sqlite

### Issue
Forgot enableSemantics() in my LocalSettings when working on a PF issue and got the following when PF invoked SMW. Adding enableSemantics results in the better error about a missing upgrade key.
```
Exception encountered, of type "RuntimeException"
[c5aa3545844ee55634e401c4] /core/index.php/Special:RunQuery/Hydro_lesson_query RuntimeException from line 70 of /home/mah/work/code/mediawiki/core/extensions/SemanticMediaWiki/src/Connection/ConnectionManager.php: mw.db is missing a registered connection provider
```
### Backtrace
```
#0 /../core/extensions/SemanticMediaWiki/src/Connection/ConnectionManager.php(32): SMW\Connection\ConnectionManager->findConnectionProvider(string)
#1 /../core/extensions/SemanticMediaWiki/src/Store.php(598): SMW\Connection\ConnectionManager->getConnection(string)
#2 /../core/extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_SQLStore3.php(627): SMW\Store->getConnection(string)
#3 /../core/extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_Sql3SmwIds.php(285): SMWSQLStore3->getConnection(string)
#4 /../core/extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_Sql3SmwIds.php(854): SMWSql3SmwIds->getDatabaseIdAndSort(string, string, string, string, string, boolean, boolean)
#5 /../core/extensions/SemanticMediaWiki/src/SQLStore/EntityStore/EntityLookup.php(301): SMWSql3SmwIds->getSMWPropertyID(SMW\DIProperty)
#6 /../core/extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_SQLStore3.php(190): SMW\SQLStore\EntityStore\EntityLookup->getPropertyValues(NULL, SMW\DIProperty, SMW\RequestOptions)
#7 /../core/extensions/PageForms/includes/PF_ValuesUtils.php(32): SMWSQLStore3->getPropertyValues(NULL, SMW\DIProperty, SMW\RequestOptions)
#8 /../core/extensions/PageForms/includes/PF_ValuesUtils.php(123): PFValuesUtils::getSMWPropertyValues(SMWSQLStore3, NULL, string, SMW\RequestOptions)
#9 /../core/extensions/PageForms/includes/PF_FormField.php(273): PFValuesUtils::getAllValuesForProperty(string)
#10 /../core/extensions/PageForms/includes/PF_FormPrinter.php(1010): PFFormField::newFromFormFieldTag(array, PFTemplate, PFTemplateInForm, boolean)
#11 /../core/includes/StubObject.php(105): PFFormPrinter->formHTML(string, boolean, boolean, integer, NULL, NULL, NULL, boolean, boolean)
#12 /../core/includes/StubObject.php(129): StubObject->_call(string, array)
#13 /../core/extensions/PageForms/specials/PF_RunQuery.php(79): StubObject->__call(string, array)
#14 /../core/extensions/PageForms/specials/PF_RunQuery.php(26): PFRunQuery->printPage(string, boolean)
#15 /../core/includes/specialpage/SpecialPage.php(479): PFRunQuery->execute(string)
#16 /../core/includes/specialpage/SpecialPageFactory.php(577): SpecialPage->run(string)
#17 /../core/includes/MediaWiki.php(282): SpecialPageFactory::executePath(Title, RequestContext)
#18 /../core/includes/MediaWiki.php(735): MediaWiki->performRequest()
#19 /../core/includes/MediaWiki.php(509): MediaWiki->main()
#20 /../core/index.php(43): MediaWiki->run()
#21 {main}
```

### PR DESCRIPTION
**Pull request by kghbln**
This PR is made in reference to: #4150 

This PR addresses or contains:
- Improves error handling

This PR includes:
- [ ] Tests (unit/integration)
- [ ] CI build passed

Fixes #4150

### CODE REVIEW

File: src/Connection/ConnectionManager.php [hunk 1]
```diff
@@ -67,6 +67,15 @@ private function findConnectionProvider( $id ) {
 			return self::$connectionProviders[$id];
 		}
 
+		// #4150
+		if ( !defined( 'SMW_VERSION' ) ) {
+			$setupCheck = new SetupCheck();
```
Comment (mwjames) [inline src/Connection/ConnectionManager.php]: Have a look the posted change, see `use SMW\SetupCheck`.

Comment (kghbln) [reply src/Connection/ConnectionManager.php]: Thanks for the pointer. I just have too many windows open so I pretty messed :(

**Comment by mwjames** [comment]
@hexmode Please confirm that this works as expected when `enableSemantics` isn't used and PF tries to access SMW.

**Comment by hexmode** [comment]
This does not work.

First, I had to modify ConnectionManager.php:
```
@@ -70,7 +70,11 @@ class ConnectionManager {
 
 		// #4150
 		if ( !defined( 'SMW_VERSION' ) ) {
-			$setupCheck = new SetupCheck();
+			$setupCheck = new SetupCheck(
+				[
+					'wgScriptPath' => $GLOBALS['wgScriptPath'],
+				]
+			);
 
```

Then, no messages displayed because the i18n messages for SMW weren't available.  I poked around in the code trying different things, but no matter what, I still couldn't get the message.

So, good start, but not there yet.

**Comment by mwjames** [comment]
> Then, no messages displayed because the i18n messages for SMW weren't
> available.  I poked around in the code trying different things, but no
> matter what, I still couldn't get the message.
>
> So, good start, but not there yet.

I'll try to a have look at it next week.

**Comment by mwjames** [comment]
> I'll try to a have look at it next week.

So, for better or for worse, I made the change.

> Then, no messages displayed because the i18n messages for SMW weren't available. I poked around in the code trying different things, but no matter what, I still couldn't get the message.

So, yes, because those weren't registered therefore we display a canonical text that is maintained in the `SetupCheck` file without relying on an active i18n for this particular case.

Trying to access SMW (or better its store) without being properly enabled would produce something like as shown below. The display contains a stack trace therefore is should be fairly simple to find out which program/extension called SMW.

image

### Example

As indicated by `LocalSettings.php(203)` it called `smwfGetStore()->getSemanticData( \SMW\DIWIkiPage::newFromText( 'Foo' ) );` as test.

```
#0 ...\extensions\SemanticMediaWiki\src\Connection\ConnectionManager.php(42): SMW\Connection\ConnectionManager->isConnectable()
#1 ...\extensions\SemanticMediaWiki\src\Store.php(598): SMW\Connection\ConnectionManager->getConnection('mw.db')
#2 ...\extensions\SemanticMediaWiki\includes\storage\SQLStore\SMW_SQLStore3.php(627): SMW\Store->getConnection('mw.db')
#3 ...\extensions\SemanticMediaWiki\includes\storage\SQLStore\SMW_Sql3SmwIds.php(285): SMWSQLStore3->getConnection('mw.db')
#4 ...\extensions\SemanticMediaWiki\includes\storage\SQLStore\SMW_Sql3SmwIds.php(260): SMWSql3SmwIds->getDatabaseIdAndSort('Foo', 0, '', '', '', true, true)
#5 ...\extensions\SemanticMediaWiki\src\SQLStore\EntityStore\EntityLookup.php(86): SMWSql3SmwIds->getSMWPageIDandSort('Foo', 0, '', '', '', true, true)
#6 ...\extensions\SemanticMediaWiki\includes\storage\SQLStore\SMW_SQLStore3.php(176): SMW\SQLStore\EntityStore\EntityLookup->getSemanticData(Object(SMW\DIWikiPage), false)
#7 ...\LocalSettings.php(203): SMWSQLStore3->getSemanticData(Object(SMW\DIWikiPage))
#8 ...\includes\Setup.php(103): require_once('C:\\xampp\\htdocs...')
#9 ...\includes\WebStart.php(77): require_once('C:\\xampp\\htdocs...')
#10 ...\index.php(39): require('C:\\xampp\\htdocs...')
#11 {main}
```

**Comment by mwjames** [comment]
PS: Please review the changes as a matter of self-study to reduce the reliance on me as developer capacity.

**Comment by hexmode** [comment]
Thanks for helping out.  I was tempted to just put in the raw English text, but I thought there might be a way to initialize the i18n system to show the proper messages.  The backtrace is a good touch, too.

**Pull request merged**